test/topology_custom/test_gossiper_orphan_remover.py

```diff
@@ -0,0 +1,39 @@
+import time
+import pytest
+import logging
+from test.pylib.manager_client import ManagerClient
+from test.topology.conftest import skip_mode
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_crashed_node_substitution(manager: ManagerClient):
+    servers = await manager.servers_add(3, config={
+        'error_injections_at_startup': ['fast_orphan_removal_fiber']
+    })
+
+    await manager.server_add(config={
+        'error_injections_at_startup': ['crash_before_group0_join']
+    }, expected_error="deliberately crashed for orphan remover test")
+
+    live_eps = await manager.api.client.get_json("/gossiper/endpoint/live", host=servers[0].ip_addr)
+    down_eps = await manager.api.client.get_json("/gossiper/endpoint/down", host=servers[0].ip_addr)
+
+    gossiper_eps = live_eps + down_eps
+    post_crash_servers = await manager.running_servers()
+    server_eps = [s.ip_addr for s in post_crash_servers]
+
+    assert len(gossiper_eps) == (len(server_eps)+1)
+
+    orphan_ip = [ip for ip in gossiper_eps if ip not in server_eps][0]
+
+    [await manager.api.enable_injection(s.ip_addr, 'speedup_orphan_removal', one_shot=False, parameters={"orphan_ip": orphan_ip}) for s in servers]
+
+    time.sleep(1)  # Wait for topology coordinator to commit, and nodes to apply new changes.
```

Comment:
Still timing dependent and can result in flakiness. As the comment says, you're waiting for a specific event -- so why not wait for it explicitly, instead of sleeping for arbitrary amount of time and hoping it had happened by the time you finished sleeping?

Reply:
Done